lib/screens/v2/screen_audio_data.ex

```diff
@@ -1,25 +1,54 @@
 defmodule Screens.V2.ScreenAudioData do
   @moduledoc false
 
+  alias Screens.Config.Screen
+  alias Screens.Config.V2.{Audio, BusShelter}
   alias Screens.V2.ScreenData
   alias Screens.V2.WidgetInstance
 
   @type screen_id :: String.t()
 
-  @spec by_screen_id(screen_id()) :: list({module(), map()})
+  @spec by_screen_id(screen_id()) :: list({module(), map()}) | :error
   def by_screen_id(
         screen_id,
         get_config_fn \\ &ScreenData.get_config/1,
         fetch_data_fn \\ &ScreenData.fetch_data/2
       ) do
     config = get_config_fn.(screen_id)
+    now = DateTime.utc_now()
 
-    screen_id
-    |> fetch_data_fn.(config)
-    |> elem(1)
-    |> Map.values()
-    |> Enum.filter(&WidgetInstance.audio_valid_candidate?/1)
-    |> Enum.sort_by(&WidgetInstance.audio_sort_key/1)
-    |> Enum.map(&{WidgetInstance.audio_view(&1), WidgetInstance.audio_serialize(&1)})
+    case config do
+      %Screen{app_params: %app{}} when app not in [BusShelter] ->
+        :error
+
+      %Screen{app_params: %_app{audio: nil}} ->
+        :error
+
+      %Screen{app_params: %_app{audio: audio}} ->
+        if date_out_of_range?(audio, now) do
+          [%{}]
+        else
+          screen_id
+          |> fetch_data_fn.(config)
+          |> elem(1)
+          |> Map.values()
+          |> Enum.filter(&WidgetInstance.audio_valid_candidate?/1)
+          |> Enum.sort_by(&WidgetInstance.audio_sort_key/1)
+          |> Enum.map(&{WidgetInstance.audio_view(&1), WidgetInstance.audio_serialize(&1)})
+        end
+    end
+  end
+
+  defp date_out_of_range?(
+         %Audio{
+           start_time: start_time,
+           stop_time: stop_time,
+           days_active: days_active
+         },
+         now
+       ) do
+    Date.day_of_week(now) not in days_active or
+      DateTime.compare(start_time, now) == :gt or
+      DateTime.compare(stop_time, now) == :lt
```

Comment:
`now` is in UTC timezone, so you'll need to shift it to Eastern time before checking against the values from configuration which we're also expecting to be in Eastern time, in order for DST changes to take effect automatically in the readout schedule.

You'll also want to use `Time.compare` instead of `DateTime.compare` since we're only interested in the time of day, not the specific date.
```ex
{:ok, now_eastern} = DateTime.shift_zone(now, "America/New_York")

Date.day_of_week(now_eastern) ...
Time.compare(now_eastern, ...) == ...
```